demos/coreHTTP/http_demo_s3_download_multithreaded.c

```diff
@@ -0,0 +1,965 @@
+            xStatus = pdFAIL;
+        }
+    }
+
+    /* Parse the Content-Range header value to get the file size. */
+    if( xStatus == pdPASS )
+    {
+        pcFileSizeStr = strstr( pcContentRangeValStr, "/" );
+
+        if( pcFileSizeStr == NULL )
+        {
+            LogError( ( "'/' not present in Content-Range header value: %s.",
+                        pcContentRangeValStr ) );
+            xStatus = pdFAIL;
+        }
+    }
+
+    if( xStatus == pdPASS )
+    {
+        pcFileSizeStr += sizeof( char );
+        *pxFileSize = ( size_t ) strtoul( pcFileSizeStr, NULL, 10 );
+
+        if( ( *pxFileSize == 0 ) || ( *pxFileSize == UINT32_MAX ) )
+        {
+            LogError( ( "Error using strtoul to get the file size from %s: xFileSize=%d.",
+                        pcFileSizeStr, ( int32_t ) *pxFileSize ) );
+            xStatus = pdFAIL;
+        }
+    }
+
+    if( xStatus == pdPASS )
+    {
+        LogInfo( ( "The file is %d bytes long.", ( int32_t ) *pxFileSize ) );
+    }
+
+    return xStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvStartHTTPThread( void * pvArgs )
+{
+    HTTPStatus_t xHTTPStatus = HTTPSuccess;
+    BaseType_t xStatus = pdPASS;
+    /* The transport layer interface used by the HTTP Client library. */
+    TransportInterface_t xTransportInterface;
+
+    ( void ) pvArgs;
+
+    /* Define the transport interface. */
+    xTransportInterface.pNetworkContext = &xNetworkContext;
+    xTransportInterface.send = SecureSocketsTransport_Send;
+    xTransportInterface.recv = SecureSocketsTransport_Recv;
+
+    /* Initialize response struct. */
+    xResponseItem.xResponse.pBuffer = xResponseItem.ucResponseBuffer;
+    xResponseItem.xResponse.bufferLen = democonfigUSER_BUFFER_LENGTH;
+
+    for( ; ; )
+    {
+        /* Read request from queue. */
+        xStatus = xQueueReceive( xRequestQueue,
+                                 &xRequestItem,
+                                 httpexampleDEMO_TICKS_TO_WAIT );
+
+        if( xStatus == pdFAIL )
+        {
+            LogInfo( ( "No requests in the queue. Trying again." ) );
+            continue;
+        }
+
+        LogInfo( ( "HTTP thread retrieved request." ) );
+        LogDebug( ( "Request Headers:\n%.*s",
+                    ( int32_t ) xRequestItem.xRequestHeaders.headersLen,
+                    ( char * ) xRequestItem.xRequestHeaders.pBuffer ) );
+
+        xHTTPStatus = HTTPClient_Send( &xTransportInterface,
+                                       &xRequestItem.xRequestHeaders,
+                                       NULL,
+                                       0,
+                                       &xResponseItem.xResponse,
+                                       0 );
+
+        if( xHTTPStatus != HTTPSuccess )
+        {
+            LogError( ( "Failed to send HTTP request: Error=%s.",
+                        HTTPClient_strerror( xHTTPStatus ) ) );
```

Comment:
How would the main task know if there is an error?

Reply:
Originally, only if a corresponding response to an outgoing request was not present in the response queue. But I have updated the HTTP task to send a notification to the main task on HTTP failure.